pkg/custom-provider/provider.go

```diff
@@ -302,17 +285,49 @@ func (l *cachingMetricsLister) RunUntil(stopChan <-chan struct{}) {
 func (l *cachingMetricsLister) updateMetrics() error {
 	startTime := pmodel.Now().Add(-1 * l.updateInterval)
 
-	sels := l.Selectors()
+	// don't do duplicate queries when it's just the matchers that change
+	seriesCacheByQuery := make(map[prom.Selector][]prom.Series)
```

Comment:
This map has concurrency issues. Getting `concurrent map writes` and `concurrent map read and map write`.

Would it be worth using a `syncmap` for this? I've built locally and wrapped all uses within a mutex locak and unlock cycle, seems to fix the issue.